collection-scripts/gather_trigger_gmr

```diff
@@ -45,6 +48,56 @@ cinder_trigger_gmr() {
 }
 
 
+nova_trigger_gmr() {
+    # https://docs.openstack.org/nova/latest/reference/gmr.html
+    echo "Trigger GMR for Nova services"
+
+    schedulers=`$oc get pod -l service=nova-scheduler -o=custom-columns=N:.metadata.name --no-headers | tr '\n' ' '`
+    for scheduler in $schedulers
+    do
+        run_bg $oce $scheduler -- pkill --signal USR2 nova-scheduler
+    done
+
+    conductors=`$oc get pod -l service=nova-conductor -o=custom-columns=N:.metadata.name --no-headers | tr '\n' ' '`
+    for conductor in $conductors
+    do
+        run_bg $oce $conductor -- pkill --signal USR2 nova-conductor
+    done
+
+    novncs=`$oc get pod -l service=nova-novncproxy -o=custom-columns=N:.metadata.name --no-headers | tr '\n' ' '`
+    for novnc in $novncs
+    do
+        run_bg $oce $novnc -- pkill --signal USR2 nova-novncproxy
+    done
+
+    apis=`$oc get pod -l service=nova-api -o=custom-columns=N:.metadata.name --no-headers | tr '\n' ' '`
+    for api in $apis
+    do
+        # The nova-api uses apache and apache disables signal handling so
+        # we use a file trigger.
+        run_bg $oce -c nova-api-api $api -- touch /var/log/nova/gmr_trigger
+        # NOTE(gibi): This is a hack. The nova-api uses eventlet.
+        # All threads including the GMR file monitoring one are frozen by
+        # apache if there is no active request to handle. The requests from
+        # the k8s probes are not enough as they are handled without any
+        # eventlet yielding. So the GMR monitoring thread is not woken up.
+        # Therefore we are sending an API request that needs DB access as such
+        # access will involve an eventlet yield letting the GMR monitoring
+        # thread a chance to run and detect the trigger.
+        # If we have multiple nova-api replicas we are relying on the assumed
+        # round robin scheduling of the k8s service to the backends to send
+        # an API request to each nova-api pod in this loop.
+        # NOTE(gibi): The GMR report will be emitted not to the normal API log
+        # but the log of the apache process but must gather pulls both
+        # containers' log so we are good.
+        run_bg $oce openstackclient -- openstack compute service list
```

Comment:
Shouldn´t we first touch all the files and then do the API requests to leverage the round robin?

As it is now the API request could be going to a nova-api that hasn´t had the file touched yet.

Reply:
good point

Reply:
done